addons/im_livechat/models/discuss_channel.py

```diff
@@ -241,7 +241,7 @@ def _message_post_after_hook(self, message, msg_vals):
                 )
                 question_msg.user_script_answer_id = selected_answer
                 if store := self.env.context.get("message_post_store"):
-                    store.add(question_msg.mail_message_id)
+                    store.add(message, for_current_user=True).add(question_msg.mail_message_id)
```

Comment:
JS handling of return of message post takes the first message of store and adds it to thread.
We know this is a weakness and we should return the specific id to insert and never rely on ordering.

The fix is just to ensure the answer is the first message added to store, and not the question (which is already in message list).

Reply:
Badly created override in message post due to doing in stable forces a "repeat" of the `.add(message, for_current_user=True)` that is already present in original controller.